source/windows.js

```diff
@@ -21,18 +21,21 @@ const shouldForceShell = async (file, {shell, cwd, env = process.env}) => proces
 // Windows allows omitting file extensions (present in the `PATHEXT` environment variable).
 // Therefore we must use the `PATH` environment variable and make `access` calls to check this.
 // Environment variables are case-insensitive on Windows, so we check both `PATH` and `Path`.
-// eslint-disable-next-line no-return-assign
 const isExe = (file, cwd, {Path = '', PATH = Path}) =>
 	// If the *.exe or *.com file extension was not omitted.
 	// Windows common file systems are case-insensitive.
 	exeExtensions.some(extension => file.toLowerCase().endsWith(extension))
-	// Use returned assignment to keep code small
-	|| (EXE_MEMO[`${file}\0${cwd}\0${PATH}`] ??= mIsExe(file, cwd, PATH));
+	|| mIsExe(file, cwd, PATH);
 
-// Memoize the following function, for performance
+// Memoize the `mIsExe` and `fs.access`, for performance
 const EXE_MEMO = {};
+// eslint-disable-next-line no-return-assign
+const memoize = function_ => (...arguments_) =>
+	// Use returned assignment to keep code small
+	EXE_MEMO[arguments_.join('\0')] ??= function_(...arguments_);
```

Comment:
We probably should use a different cache object for each memoized function.
The cache key cannot be collide here with those two memoized functions since they have a different arity, but just in case we were to re-use that `memoize()` helper in the future.

```js
// Memoize the `mIsExe` and `fs.access`, for performance
const memoize = function_ => {
  const EXE_MEMO = {}
  // eslint-disable-next-line no-return-assign
  return (...arguments_) =>
	  // Use returned assignment to keep code small
	  EXE_MEMO[arguments_.join('\0')] ??= function_(...arguments_);
}
```

Reply:
Since it's "nano" 😄

Reply:
Ok, but let's keep this in mind.